src/idl_gen_general.cpp

```diff
@@ -587,6 +587,19 @@ std::string GenGetter(const Type &type) {
   }
 }
 
+// Returns the function name that is able to read a value of the given type.
+std::string GenGetterForLookupByKey(const Type &type,
+                                    const std::string data_buffer) {
+  auto getter = data_buffer + "." + FunctionStart('G') + "et";
+  if (type.base_type == BASE_TYPE_BOOL) {
+    getter = "0!=" + getter;
+  }
+  else if (GenTypeBasic(type, false) != "byte") {
```

Comment:
There are problems with `uint`, `ubyte` keys in Java that are greater than positive maximum of `int`, `byte`. 
I have fixed it, but wanted to ask before pushing:
Should key getter for `uint` keys in Java generate something like this:
long val_1 = (long)bb.getInt(__offset(4, o1, _bb)) & 0xFFFFFFFFL ?